irksome/stage_value.py

```diff
@@ -186,7 +188,24 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
         self.appctx["stage_type"] = "value"
         self.appctx["vandermonde"] = vandermonde
 
-        if (not butcher_tableau.is_stiffly_accurate) and (basis_type != "Bernstein"):
+        if use_collocation_update:
+            # Use the terminal value of the collocation polynomial to update the solution. Note: collocation update is only implemented for constant-in-time boundary conditions.
+            # TODO: create an assertion to check for constant-in-time boundary conditions.
+            nodes = numpy.insert(self.butcher_tableau.c, 0, 0.0)
+
+            assert isinstance(self.butcher_tableau, CollocationButcherTableau), "Need a collocation method for collocation update"
+            assert (self.basis_type is None or self.basis_type == "Lagrange"), "Collocation update requires the Lagrange form of the collocation polynomial"
+            assert (len(set(nodes)) == self.butcher_tableau.num_stages + 1), "Need a non-confluent collocation method to use collocation update"
+
+            lag_basis = LagrangePolynomialSet(ufc_simplex(1), nodes)
+            collocation_vander = vecconst(lag_basis.tabulate((1.0,))[(0,)])
+            stage_vals = numpy.insert(self.stages.subfunctions, 0, [self.u0.subfunctions[nf] for nf in range(self.num_fields)])
+
+            self.collocation_vander = collocation_vander
+            self.stage_vals = stage_vals
```

Comment:
Maybe we could get rid of `self.stage_vals` and construct the tuple each time step
```suggestion
            self.collocation_vander = collocation_vander
            self.stage_vals = self.u0.subfunctions + self.stages.subfunctions
```

Reply:
Agreed. This cleans things up nicely.